prompt/select.ts

```diff
@@ -14,10 +14,15 @@ import { GenericPrompt } from "./_generic_prompt.ts";
 export type SelectKeys = GenericListKeys;
 
 /** Select option options. */
-export type SelectOption = GenericListOption;
+export interface SelectOption extends GenericListOption {
+  options?: SelectValueOptions;
+}
```

Comment:
I mean, the user should be able to add an option or a group of options but not a group with a value.
So the `SelectValueOptions` type could be updated like this:
```ts
export type SelectValueOptions = (string | SelectOption | SelectOptionGroup)[];
```
But I think we would already need to add support for nested options to the `GenericList` class for this 🤔

Reply:
I'm having trouble making the compiler happy here at least, so I suspect you're right. But I'm having trouble grokking how all the types line up, and what exactly it is that makes `SelectOptionGroup` incompatible with the existing code.

Reply:
Before I start down this path, I'd like to hear more from you about it. It seems like this could lead to a lot of scope creep, if it leads to a bunch of non-`Select` controls also supporting groups. Is that desirable? Can we implement this support in `GenericList`, but only expose it in `Select` for now? And is it even what you think should happen?

Reply:
Sry for late reply! Than lets keep it  as it is for now. We can adapt the interfaces and the checkbox prompt in a second PR.

Reply:
I have now moved the logic to the generic list class and also splitted the option interfaces. So options have now the type `Array<string | SelectOption | SelectOptionGroup>`.

Reply:
I forgot that it is not possible to use interfaces as union type as long as they don't have overlapping properties with a different type. I think the only solution is to re-use the `value` property for the child options insted of the `options` property.